server/src/service/service.rs

```diff
@@ -0,0 +1,135 @@
+        let session = Session::new(ObjectPath::try_from("/s1").unwrap().into()).await;
+        Ok((
+            OwnedValue::try_from(input).unwrap(),
+            ObjectPath::try_from(session.path().await).unwrap().into(),
+        ))
+    }
+
+    pub async fn create_collection(
+        &self,
+        #[zbus(object_server)] object_server: &ObjectServer,
+        properties: HashMap<&str, &str>,
+        alias: String,
+    ) {
+        let collection = Collection::new(alias.clone()).await;
+        let path = format!("/org/freedesktop/secrets_/collection/{}", &alias);
+
+        object_server.at(path, collection);
+    }
+
+    pub async fn search_items(&self, attributes: HashMap<&str, &str>) {
+        let items = match KEYRING.get().unwrap().search_items(attributes).await {
+            Ok(i) => i,
+            Err(_) => todo!(),
+        };
+
+        let mut unlocked: Vec<Item>;
+        let mut locked: Vec<Item>;
+
+        for item in items {
+            // TODO
+            // needs a way to access item's locked or unlocked status
+        }
+    }
+
+    pub async fn unlock(&self, objects: Vec<OwnedObjectPath>) {
+        // TODO
+        /*
+         * loop through provided objects accessing locked state
+         * if it's already locked unlock
+         * otherwise do nothing
+         */
+    }
+
+    pub async fn lock(&self, objects: Vec<OwnedObjectPath>) {
+        // TODO
+        // opposite impl of unlock
+    }
+
+    pub async fn get_secrets(&self, items: Vec<OwnedObjectPath>) {
+        let n_secrets = KEYRING.get().unwrap().n_items().await;
+        if n_secrets == 0 {
+            return;
+        }
+    }
+
+    pub async fn read_alias(&self, name: &str) -> ObjectPath {
+        let mut ret = ObjectPath::try_from("/").unwrap();
+        for collection in &self.collections {
+            if collection.label().await == name {
+                ret = collection.path().await;
+            }
+        }
+        ret
+    }
+
+    pub async fn set_alias(&self, name: String, collection: OwnedObjectPath) {
+        for collection_ in self.collections().await {
+            if collection_.path().await.as_str() == collection.as_str() {
+                //collection_.set_label(name.clone()).await;
+            }
+        }
+    }
+
+    /*
+    #[dbus_interface(property)]
+     pub async fn get_collections(&self) -> &Vec<Collection> {
+         &self.collections
+     }*/
+
+    #[dbus_interface(signal)]
+    pub async fn collection_created(ctxt: &SignalContext<'_>) -> Result<(), Error>;
+
+    #[dbus_interface(signal)]
+    pub async fn collection_deleted(ctxt: &SignalContext<'_>) -> Result<(), Error>;
+
+    #[dbus_interface(signal)]
+    pub async fn collection_changed(ctxt: &SignalContext<'_>) -> Result<(), Error>;
+}
+
+impl Service {
+    pub async fn collections(&self) -> &Vec<Collection> {
+        &self.collections
+    }
+}
+
+pub async fn init_secret_service() -> Service {
+    let mut service = Service {
```

Comment:
`mut` is not necessary here. You do not change the value.

I'd recommend adding the `unused_mut` clippy lint :)

Reply:
Yeah, I agree.